internal/cmd/launcher_test.go

```diff
@@ -348,3 +350,64 @@ func TestIsAnalysisRequired(t *testing.T) {
 		})
 	}
 }
+
+func TestGetBuildServiceURL(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name                      string
+		buildSrvURL               string
+		enableCommunityExtensions bool
+		expectErr                 bool
+		expected                  string
+	}{
+		{
+			name:                      "default build service url",
+			buildSrvURL:               state.DefaultBuildServiceURL,
+			enableCommunityExtensions: false,
+			expectErr:                 false,
+			expected:                  fmt.Sprintf("%s/%s", state.DefaultBuildServiceURL, state.CloudExtensionsCatalog),
+		},
+		{
+			name:                      "enable community extensions",
+			buildSrvURL:               state.DefaultBuildServiceURL,
+			enableCommunityExtensions: true,
+			expectErr:                 false,
+			expected:                  fmt.Sprintf("%s/%s", state.DefaultBuildServiceURL, state.CommunityExtensionsCatalog),
+		},
+		{
+			name:                      "invalid buildServiceURL",
+			buildSrvURL:               "https://host:port",
+			enableCommunityExtensions: false,
+			expectErr:                 true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			logger := &logrus.Logger{
+				Out: io.Discard,
+			}
+
+			flags := state.GlobalFlags{
+				BinaryProvisioning:        true,
+				BuildServiceURL:           tc.buildSrvURL,
+				EnableCommunityExtensions: tc.enableCommunityExtensions,
+			}
+
+			buildSrvURL, err := getBuildServiceURL(flags, logger)
+
+			if !tc.expectErr && err != nil {
+				t.Fatalf("unexpected error %v", err)
+			}
+
+			if tc.expectErr && err == nil {
+				t.Fatalf("expected error got none")
+			}
+
+			if !tc.expectErr && buildSrvURL != tc.expected {
+				t.Fatalf("expected buildSrvURL %q got %q", tc.expected, buildSrvURL)
+			}
```

Comment:
Nit; I think we generally use `assert` or `require` from [testify](https://github.com/stretchr/testify) 🤔